src/mapgen/mapgen_valleys.cpp

```diff
@@ -529,48 +425,37 @@ int MapgenValleys::generateTerrain()
 			index_3d += ystride;
 		}
 
-		if (heightmap[index_2d] == -MAX_MAP_GENERATION_LIMIT) {
-			s16 surface_y_int = myround(surface_y);
-
-			if (surface_y_int > node_max.Y + 1 ||
-					surface_y_int < node_min.Y - 1) {
-				// If surface_y is outside the chunk, it's good enough
-				heightmap[index_2d] = surface_y_int;
-			} else {
-				// If the ground is outside of this chunk, but surface_y is
-				// within the chunk, give a value outside.
-				heightmap[index_2d] = node_min.Y - 2;
-			}
-		}
-
 		// Optionally increase humidity around rivers
 		if (spflags & MGVALLEYS_HUMID_RIVERS) {
+			// Compensate to avoid increasing average humidity
+			m_bgen->humidmap[index_2d] *= 0.8f;
```

Comment:
Where do these `0.8` come from?

Reply:
This comes from duane's code: https://github.com/minetest/minetest/blob/1511086/src/mapgen/mapgen_valleys.cpp#L305. He implemented an increase of humidity around rivers and seas, and decreased the humidity a bit to compensate. I don't know precisely where this comes from, but it's most likely an arbitrary value.

Reply:
Yes just a value chosen through tuning, was also present in previous mgvalleys code (but in a different place).